appengine-java11/springboot-helloworld/pom.xml

```diff
@@ -94,8 +94,10 @@
         <artifactId>appengine-maven-plugin</artifactId>
         <version>2.4.1</version>
         <configuration>
-          <projectId>GCLOUD_CONFIG</projectId>
```

Comment:
My memo. Java 8 sample explains this value

> Note: GCLOUD_CONFIG is a special version for autogenerating an App Engine version. Change this field to specify a specific version name.

https://github.com/GoogleCloudPlatform/java-docs-samples/tree/main/appengine-java8/springboot-helloworld

Reply:
Thanks a lot for your reviews.

From reading the doc thoroughly, it seems I misunderstood and it may be the other way around.
This project is correct as specified by the doc :

> projectId -->	The Google Cloud Platform project name to use for this invocation. You must specify a projectId or you can set GCLOUD_CONFIG to use the project set in your gcloud config state.

While https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/appengine-java8/springboot-helloworld/pom.xml is using **GCLOUD_CONFIG** in `version` which isn't mentioned in the doc.

> version -->	The version of the app that will be created or replaced by this deployment. If you do not specify a version, one will be generated for you.

Should I close this PR and open a new one for the java8 project ?

Reply:
In fact reference from java11 https://cloud.google.com/appengine/docs/standard/java11/maven-reference is not referring to `GCLOUD_CONFIG` value.

Reply:
The curreny pom.xml comment was confusing to me as well. I'll have my teammate to clarify these configuration.

You can close this pull request. Thank you for pointing out the confusion.

Reply:
I don't think there's anything inherently wrong with the current samples. They are just showing different possible combinations, whether or not it's intended. I think `GCLOUD_CONFIG` is a good choice in a sample, which will make the appengine-mave-plugin retrieve the project ID configured by `gcloud`.

But I'm not sure if the following doc is accurate:
```
GCLOUD_CONFIG is a special version for autogenerating an App Engine version.
```
Check this [doc](https://github.com/GoogleCloudPlatform/app-maven-plugin/blob/master/USER_GUIDE.md#deploy).

Reply:
Thank you for the reference. It explains that GCLOUD_CONFIG is valid for both places.

<img width="1188" alt="Screen Shot 2021-12-01 at 10 06 18 AM" src="https://user-images.githubusercontent.com/28604/144259542-35d92204-c954-40a1-8adf-9bd598fe60f5.png">